### PR TITLE
DOC: Remove reference to basic RNG

### PR DESCRIPTION
**Pull request by bashtage**
Remove reference to outdated basic RNG

### CODE REVIEW

File: doc/source/reference/random/extending.rst [hunk 1]
```diff
@@ -70,9 +70,9 @@ directly from the ``_generator`` shared object, using the `BitGenerator.cffi` in
     :start-after: dlopen
 
 
-New Basic RNGs
-==============
-`~Generator` can be used with other user-provided BitGenerators. The simplest
+New Bit Generators
+==================
+`~Generator` can be used with other user-provided `~BitGenerators`. The simplest
 way to write a new BitGenerator is to examine the pyx file of one of the
```
Comment (mattip) [inline doc/source/reference/random/extending.rst]: The `s` on `BitGenerators` will [not link](https://10736-908607-gh.circle-artifacts.com/0/home/circleci/repo/doc/build/html/reference/random/extending.html#new-bit-generators) properly. Maybe "with any `BitGenerator`"

Comment (eric-wieser) [reply doc/source/reference/random/extending.rst]: Or `` `~BitGenerator`\ s``

Comment (bashtage) [reply doc/source/reference/random/extending.rst]: I used @eric-wieser's suggestion

**Review by bashtage — COMMENTED** [review]
\ s

**Comment by mattip** [comment]
Thanks @bashtage

**Pull request merged**